src/commands/utils/uploadToProvider.ts

```diff
@@ -2,7 +2,66 @@ import { logError } from "../../errors";
 import { Logger } from "../../logging";
 import { showErrorNotificationWithButtons } from "../../notifications";
 
-const logger = new Logger("commands/utils/uploadToAWS");
+const logger = new Logger("commands/utils/uploadToProvider");
+
+export async function uploadFileToAzure({
+  file,
+  presignedUrl,
+  contentType,
+}: {
+  file: File | Blob;
+  onUploadProgress?: (percentageDec: number) => void;
+  presignedUrl: string;
+  contentType: string;
+}): Promise<Response> {
+  logger.info("Starting Azure file upload", {
+    fileSize: file.size,
+    contentType,
+    presignedUrlHost: new URL(presignedUrl).host,
+  });
+
+  try {
+    const response = await fetch(presignedUrl, {
+      method: "PUT",
+      headers: {
+        "Content-Type": contentType,
+        "x-ms-blob-type": "BlockBlob",
+      },
+      body: file,
+    });
+
+    if (!response.ok) {
+      const responseText = await response.text();
+      logger.error("Azure upload failed", {
+        status: response.status,
+        statusText: response.statusText,
+        responseBody: responseText,
+        headers: Object.fromEntries(response.headers.entries()),
+      });
+      throw new Error(`Azure upload failed: ${response.status} ${response.statusText}`);
+    }
+
+    logger.info("Azure upload successful", {
+      status: response.status,
+      statusText: response.statusText,
+      contentLength: response.headers.get("content-length"),
+      etag: response.headers.get("etag"),
+    });
+
+    return response;
+  } catch (error) {
+    logger.error("Azure upload error", error);
```

Comment:
Minor: this isn't necessary since `logError()` also [logs the error](https://github.com/confluentinc/vscode/blob/e42ce0829151e3fd6a367950df87a7cea934019f/src/errors.ts#L188) to the console + output channel (with some extra context)